e2e/specs/onboarding-test.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { test } from '@playwright/test';
+import { HomePage } from '../pages';
+import { expect } from '@playwright/test';
+
+test('Basic Walkthrough', async ({ page }) => {
+  const homePage = new HomePage(page);
+
+  await test.step('When I visit the home page', async () => {
+    await homePage.goto();
+  });
+
+  await test.step('And I click on the help menu button', async () => {
+    await page.locator('[id="single-spa-application\\:\\@openmrs\\/esm-help-menu-app-page-0"]').getByRole('button').click();
+  });
+
+  await test.step('And I click on the tutorials button', async () => {
+    await page.getByText(/tutorials/i).click();
+  });
+
+  await test.step('Then I should see the tutorial modal', async () => {
+    await expect(page.getByRole('heading', { name: 'Tutorials' })).toBeVisible();
+    await expect(
+      page.getByText(/find walkthroughs and video tutorials on some of the core features of openMRS./i),
+    ).toBeVisible();
+  });
+
+  await test.step('And I click on the Basic Tutorial walkthrough', async () => {
+    await page.getByText('Walkthrough').nth(1).click();
```

Comment:
We can't guarantee the order of the list. The order may change when we add new tutorials. So, we need to think of a better selector.

Reply:
@jayasanka-sack I didn't figure out how we can select those walkthoroughs.
An approach that came to my mind is having a selector in the tutorial modal  [#L49](https://github.com/openmrs/openmrs-esm-user-onboarding/blob/14696d75353c201547ceeeb848d34180c3ef767a/src/tutorial/modal.component.tsx#L49) and searching for the Heading and corresponding walkthrough. WDYT?

Reply:
First of all, I've noticed a few adjustments needed for the tutorial to follow best practices:

- Change the list to an unordered list. This will be beneficial for both testing and accessibility.
- Currently, the walkthrough button is a div. It should be a button. (A ghost button would be the option.)
Here is how your code should look (there won't be any visual changes):

```
<ul>
   {tutorials.map -> tutorial}
      <li>
          <h3>{title}</h3>
          <p>{description}</p>
          <Button kind="ghost">Button</Button>
      </li>
   {/tutorials}
</ul>

```

Once you've done that, you can select the button with this:
```js
const tutorialTitle = "Basic ..."
await page.locator('li', { hasText: 'my item' }).locator('button').click();
```

Reply:
There's a small visual change in the alignment of buttons. is this fine?

https://github.com/user-attachments/assets/150196c2-d949-4c6f-b8f2-c5d888df6bd4

Reply:
That's fine. Btw, can we get the close button fixed?

Reply:
> That's fine. Btw, can we get the close button fixed?

This is due to the upgrade in carbon dependencies in this [PR](https://github.com/openmrs/openmrs-esm-form-builder/pull/309)

This would be the fix [#215](https://github.com/openmrs/openmrs-esm-home/pull/215), I'll talk with dennis regarding merge of that PR in todays coffee call.